messaging-client-databus/src/main/java/com/inmobi/databus/readers/LocalStreamCollectorReader.java

```
@@ -281,4 +281,61 @@ protected Date getTimeStampFromCollectorStreamFile(FileStatus file) {
     }
     return null;
   }
+
+  protected Long doRecursiveSizing(Path dir, PathFilter pathFilter) throws IOException {
+    Long pendingSize =0l;
+    FileStatus[] fileStatuses = fsListFileStatus(dir, pathFilter);
+    if (fileStatuses == null || fileStatuses.length == 0) {
+      LOG.debug("No files in directory:" + dir);
+    } else {
+      for (FileStatus file : fileStatuses) {
+        if (file.isDir()) {
+          doRecursiveSizing(file.getPath(), pathFilter);
+        } else {
+          try {
+            Date currentTimeStamp = LocalStreamCollectorReader.
+                    getDateFromStreamFile(streamName, file.getPath().getName());
+            if (stopTime != null && stopTime.before(currentTimeStamp)) {
+              stopListing();
+            } else {
+              pendingSize += file.getLen();
+            }
+          } catch (Exception e) {
+            e.printStackTrace();
```

handled logging as well as throwing back exception, rather than swallowing it